dashboard/test/controllers/api/v1/assessments_controller_test.rb

```diff
@@ -39,22 +50,188 @@ class Api::V1::AssessmentsControllerTest < ActionController::TestCase
+    assert_equal expected_questions, level_response["questions"]
+  end
+
   # section_responses tests - gets student responses to assessment
   test 'logged out cannot get assessment responses from students' do
     get :section_responses
     assert_response :forbidden
   end
 
+  test "don't show assessment responses to teacher who doesn't own that section" do
+    script = create :script
+    sign_in @teacher_other
+
+    get :section_responses, params: {
+      section_id: @section.id,
+      script_id: script.id
+    }
+    assert_response :forbidden
+  end
+
   test 'students cannot get assessment responses from students' do
-    sign_in @student
+    sign_in @student_1
     get :section_responses
     assert_response :forbidden
   end
 
-  test 'verified teacher can get assessment responses from students' do
+  test 'gets no assessment responses from students when no assessment' do
     sign_in @teacher
     get :section_responses, params: {section_id: @section.id, script_id: 2}
     assert_response :success
     assert_equal '{}', @response.body
   end
+
+  test "verified teacher should get assessments responses" do
+    # Sign in and create a new script.
+    sign_in @teacher
+    script = create :script
+
+    # Set up an assessment for that script.
+    sub_level1 = create :text_match, name: 'level_free_response', type: 'TextMatch'
+    sub_level2 = create :multi, name: 'level_multi_unsubmitted', type: 'Multi'
+    sub_level3 = create :multi, name: 'level_multi_correct', type: 'Multi'
+    sub_level4 = create :multi, name: 'level_multi_incorrect', type: 'Multi'
+    create :multi, name: 'level_multi_unattempted', type: 'Multi'
+
+    level1 = create :level_group, name: 'LevelGroupLevel1', type: 'LevelGroup'
+    level1.properties['title'] =  'Long assessment 1'
+    level1.properties['pages'] = [{levels: ['level_free_response', 'level_multi_unsubmitted']}, {levels: ['level_multi_correct', 'level_multi_incorrect']}, {levels: ['level_multi_unattempted']}]
+    level1.save!
+    create :script_level, script: script, levels: [level1], assessment: true
+
+    # Student has completed an assessment.
+    level_source = create(
+      :level_source,
+      level: level1,
+      data: %Q({"#{sub_level1.id}":{"result":"This is a free response"},"#{sub_level2.id}":{"result":"0"},"#{sub_level3.id}":{"result":"1"},"#{sub_level4.id}":{"result":"-1"}})
+    )
+    create :activity, user: @student_1, level: level1,
+      level_source: level_source
+
+    updated_at = Time.now
+
+    create :user_level, user: @student_1, best_result: 100, script: script, level: level1, submitted: true, updated_at: updated_at, level_source: level_source
+
+    # Call the controller method.
+    get :section_responses, params: {
+      section_id: @section.id,
+      script_id: script.id
+    }
+
+    assert_response :success
+
+    # All these are translation missing because we don't actually generate i18n files in tests.
```

Comment:
you should be able to write to a fake i18n locale to test these strings:

https://github.com/code-dot-org/code-dot-org/blob/7729b35c919ccc91119147293ae3d7e1229f54c3/dashboard/test/models/script_test.rb#L1074-L1090